src/Parser/OrientationParser.js

```diff
@@ -0,0 +1,111 @@
+    return this.setFromEuler(new THREE.Euler(-pitch, -roll, -heading, 'YXZ')); // optimized version of above
+};
+
+// From DocMicMac, the transform from local to world is:
+// RotationX(omega).RotationY(phi).RotationZ(kappa).RotationX(PI)
+// RotationX(PI) = Scale(1, -1, -1) converts between the 2 conventions for the camera local frame:
+//  X right, Y bottom, Z front : convention in webGL, threejs and computer vision
+//  X right, Y top,    Z back  : convention in photogrammetry
+THREE.Quaternion.prototype.setFromOmegaPhiKappa = function setFromOmegaPhiKappa(omega = 0, phi = 0, kappa = 0) {
+    omega *= DEG2RAD;
+    phi *= DEG2RAD;
+    kappa *= DEG2RAD;
+    this.setFromEuler(new THREE.Euler(omega, phi, kappa, 'XYZ'));
+    // this.setFromRotationMatrix(new THREE.Matrix4().makeRotationFromQuaternion(this).scale(new THREE.Vector3(1, -1, -1)));
+    this.set(this.w, this.z, -this.y, -this.x); // optimized version of above
+    return this;
+};
+
+/**
+ * @function setENUFromGeodesicNormal
+ * @param {Vector3} up - the normalized geodetic normal to the ellipsoid (given by Coordinates.geodeticNormal)
+ * @returns {Quaternion} this
+ */
+THREE.Quaternion.prototype.setENUFromGeodesicNormal = (() => {
+    const matrix = new THREE.Matrix4();
+    const elements = matrix.elements;
+    const north = new THREE.Vector3();
+    const east = new THREE.Vector3();
+    return function setENUFromGeodesicNormal(up) {
+        // this is an optimized version of matrix.lookAt(up, new THREE.Vector3(0, 0, 0), new THREE.Vector3(0, 0, 1));
+        east.set(-up.y, up.x, 0);
+        east.normalize();
+        north.crossVectors(up, east);
+        north.normalize();
+        elements[0] = east.x; elements[4] = north.x; elements[8] = up.x;
+        elements[1] = east.y; elements[5] = north.y; elements[9] = up.y;
+        elements[2] = east.z; elements[6] = north.z; elements[10] = up.z;
+        return this.setFromRotationMatrix(matrix);
+    };
+})();
+
+/**
+ *
+ * @typedef Attitude
+ * @type {Object}
+ *
+ * @property {Number} omega - angle in degrees
+ * @property {Number} phi - angle in degrees
+ * @property {Number} kappa - angle in degrees
+ * @property {Number} roll - angle in degrees
+ * @property {Number} pitch - angle in degrees
+ * @property {Number} heading - angle in degrees
+ */
+
+/**
+ * @function setFromAttitude
+ * @param {Attitude} attitude - [Attitude]{@link module:OrientedImageParser~Attitude}
+ * with properties: (omega, phi, kappa), (roll, pitch, heading) or none.
+ * @returns {THREE.Quaternion} this
+ */
+THREE.Quaternion.prototype.setFromAttitude = function setFromAttitude(attitude) {
+    if ((attitude.roll !== undefined) || (attitude.pitch !== undefined) || (attitude.heading !== undefined)) {
+        return this.setFromRollPitchHeading(attitude.roll, attitude.pitch, attitude.heading);
+    }
+    if ((attitude.omega !== undefined) || (attitude.phi !== undefined) || (attitude.kappa !== undefined)) {
```

Comment:
Specify that this is a specific case for data coming from MicMac.

Reply:
not really, eg : https://support.pcigeomatics.com/hc/en-us/articles/203483349-Heading-Pitch-Roll-vs-Omega-Phi-Kappa

Reply:
Ok thanks for the info